src/Esprima/Ast/Literal.cs

```diff
@@ -48,7 +56,7 @@ public Literal(string pattern, string flags, object? value, string raw) : this(T
     public RegexValue? Regex { [MethodImpl(MethodImplOptions.AggressiveInlining)] get; }
 
     public string? StringValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.StringLiteral ? Value as string : null; }
-    public bool BooleanValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.BooleanLiteral && NumericValue != 0; }
+    public bool BooleanValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.BooleanLiteral && ReferenceEquals(Value, _boxedTrue); }
```

Comment:
False is no boolean Value?

Reply:
On this case it should be enough to check only for true being set, like old logic checked for non-zero